pkg/healthchecker/health_checker.go

```diff
@@ -110,3 +112,64 @@ func logPatternHealthCheck(service string, logPatternsToCheck map[string]int) (b
 	}
 	return true, nil
 }
+
+// healthCheckEndpointOKFunc returns a function to check the status of an http endpoint
+func healthCheckEndpointOKFunc(endpoint string, timeout time.Duration) func() (bool, error) {
+	return func() (bool, error) {
+		httpClient := http.Client{Timeout: timeout}
+		response, err := httpClient.Get(endpoint)
+		if err != nil || response.StatusCode != http.StatusOK {
+			return false, nil
+		}
+		return true, nil
+	}
+}
+
+
+// getHealthCheckFunc returns the health check function based on the component.
+func getHealthCheckFunc(hco *options.HealthCheckerOptions) func() (bool, error) {
+	switch hco.Component {
+	case types.KubeletComponent:
+		return healthCheckEndpointOKFunc(types.KubeletHealthCheckEndpoint, hco.HealthCheckTimeout)
+	case types.KubeProxyComponent:
+		return healthCheckEndpointOKFunc(types.KubeProxyHealthCheckEndpoint, hco.HealthCheckTimeout)
+	case types.DockerComponent:
+		return func() (bool, error) {
+			if _, err := execCommand(hco.HealthCheckTimeout, getDockerPath(), "ps"); err != nil {
+				return false, nil
+			}
+			return true, nil
+		}
+	case types.CRIComponent:
+		return func() (bool, error) {
+			if _, err := execCommand(hco.HealthCheckTimeout, hco.CriCtlPath, "--runtime-endpoint="+hco.CriSocketPath, "--image-endpoint="+hco.CriSocketPath, "pods"); err != nil {
+				return false, nil
+			}
+			return true, nil
+		}
+	default:
+		glog.Warningf("Unsupported component: %v", hco.Component)
+	}
+
+	return nil
+}
+
+func getDockerPath() string {
+	if runtime.GOOS == "windows" {
```

Comment:
Given that there's already a _windows.go and _linux.go file just define the methods there.

This has 2 benefits:
1. It's clear that when a new OS is supported that this method needs to be considered for porting.
2. The logic between OSes have lower risk of leaking into each other's binaries.

Reply:
+1